### ISSUE TITLE
Need a simple way to run tests on helix, from the command line

### ISSUE THREAD

**Issue by radical** [dotnet/runtime#53829]
Right now, to debug issues with tests running on helix, you have to make changes to your PR, and wait for the build to eventually get queued, and built. That adds unnecessary time, and resources queuing, and building all the other builds too.

Instead, it would be very helpful to be able to run tests on helix, from a dev's local machine.

`aspnetcore` has something like this - https://github.com/dotnet/aspnetcore/blob/main/docs/Helix.md

cc @akoeplinger

**Comment by safern** [comment]
> Right now, to debug issues with tests running on helix, you have to make changes to your PR, and wait for the build to eventually get queued, and built. That adds unnecessary time, and resources queuing, and building all the other builds too.

> Instead, it would be very helpful to be able to run tests on helix, from a dev's local machine.

Helix is not tight to CI on dotnet/runtime either. You only need to do what CI does, build, and then just invoke `sendtohelix.proj` with the right properties and env vars set and that should do the trick. That is actually what I do to test helix changes.

**Comment by akoeplinger** [comment]
> You only need to do what CI does, build, and then just invoke sendtohelix.proj with the right properties and env vars set and that should do the trick.

Doing that is not easy or even documented though, so having a simple script which does it for you would be nice.

**Comment by ViktorHofer** [comment]
I don't see much value in a script over an msbuild invocation but I do see a ton of value in documenting the command and its parameters.

**Comment by radical** [comment]
> I don't see much value in a script over an msbuild invocation but I do see a ton of value in documenting the command and its parameters.

Maybe a msbuild target, that would work for library, and runtime tests?

**Comment by ViktorHofer** [comment]
That target already exists in sendtohelix.proj, so it probably just needs to be documented.

**Comment by safern** [comment]
> That target already exists in sendtohelix.proj, so it probably just needs to be documented.

Yes I think just document it. A script would be cumbersome as you would have to still pass down msbuild parameters... invoking MSBuild is simple.

**Comment by ericstj** [comment]
Cc @michaelgsharp who’s been working in this area in dotnet/machinelearning.

**Comment by michaelgsharp** [comment]
I did it by invoking MSBuild as @safern suggested. Since MSBuild pulls from environment variables identically to pulling from parameters, I wrote a powershell script that set the variables I needed and then called MSBuild with my submit to helix job:

```powershell
$env:BuildArchitecture="x64"
$env:HelixTargetQueues="Windows.10.Amd64.Open"
# Whatever other env variables I needed to set

powershell "$env:BUILD_SOURCESDIRECTORY\eng\common\msbuild.ps1 $env:BUILD_SOURCESDIRECTORY\eng\helix.proj /t:Test /bl:$env:BUILD_SOURCESDIRECTORY\artifacts\log\$env:BuildConfig\SendToHelix.binlog "... # Any other parameters needed go here
```

Obviously the paths will probably need to be different than mine here. I did have some issues if I had a docker image name in the HelixTargetQueues and used an environment variable, but IIRC I just switched then to passing that in with `/p` and it worked fine.

Note though that if you have slow upload internet speed and your payload is large it can take a while to upload.

**Comment by steveisok** [comment]
In the case of wasm and quite possibly iOS, we build and run the tests on helix.  If you have to make and validate a build change, it can be quite time consuming to iterate.  I think there are at least three parts to this:

1. How can we trigger a build and archive only one test suite easily?

2. Is there a target or script like @michaelgsharp specified to make it easier to send to helix?

3. What is the easiest / quickest way to browse the logs?  FWIW - I use firefox as it presents json nicely.
